src/utils/motions.ts

```diff
@@ -1,30 +1,88 @@
 import { AnyColonyClient, Extension } from '@colony/colony-js';
-import { TransactionDescription } from 'ethers/lib/utils';
+import { BigNumber } from 'ethers';
+import Decimal from 'decimal.js';
 
-import { mutate } from '../amplifyClient';
-import { formatDomain } from './domains';
-import { ContractEvent, motionNameMapping } from '../types';
-import { verbose } from './logger';
-import { getColonyTokenAddress, getColonyTokenData } from './tokens';
+import { mutate } from '~amplifyClient';
+import { formatDomain } from '~utils/domains';
+import { ContractEvent, AugmentedMotionData, motionNameMapping } from '~types';
 
-export const getParsedActionFromMotion = async (
-  motionId: string,
+import { getColonyTokenAddress } from './tokens';
+import { getVotingClient } from './clients';
+
+export const convertStakeToPercentage = (
+  stake: string | Decimal,
+  requiredStake: Decimal,
+) => new Decimal(stake).div(requiredStake).mul(100).toDP(2).toString();
+
+export const getMotionStakePercentages = (
+  yay: BigNumber,
+  nay: BigNumber,
+  requiredStake: Decimal,
+) => ({
+  yay: convertStakeToPercentage(yay.toString(), requiredStake),
+  nay: convertStakeToPercentage(nay.toString(), requiredStake),
+});
+
+export const getRequiredStake = async (
+  colonyAddress: string,
+  skillRep: string,
+) => {
+  const votingReputationClient = await getVotingClient(colonyAddress);
+  const totalStakeFraction =
+    await votingReputationClient.getTotalStakeFraction();
+  const totalStakeFractionPercentage = new Decimal(
+    totalStakeFraction.toString(),
+  ).div(new Decimal(10).pow(18).toString());
+  return new Decimal(skillRep).mul(totalStakeFractionPercentage).round();
+};
+
+export const extractDataFromMotion = async (
+  motionId: BigNumber,
   colonyClient: AnyColonyClient,
+  colonyAddress: string,
 ) => {
   const votingClient = await colonyClient.getExtensionClient(
     Extension.VotingReputation,
   );
 
-  const motion = await votingClient.getMotion(motionId);
+  const {
+    action,
+    stakes: [nay, yay],
+    rootHash,
+    domainId: motionDomainId,
+    skillRep,
+  } = await votingClient.getMotion(motionId);
 
-  try {
-    return colonyClient.interface.parseTransaction({
-      data: motion.action,
-    });
-  } catch {
-    verbose(`Unable to parse ${motion.action}`);
-    return undefined;
-  }
+  const requiredStake = await getRequiredStake(
+    colonyAddress,
+    skillRep.toString(),
+  );
+  const motionState = await votingClient.getMotionState(motionId);
+  const parsedAction = colonyClient.interface.parseTransaction({
+    data: action,
+  });
+
+  return {
+    parsedAction,
+    motionData: {
+      motionId: motionId.toString(),
+      motionStakes: {
+        raw: {
+          yay: yay.toString(),
+          nay: nay.toString(),
+        },
+        percentage: getMotionStakePercentages(yay, nay, requiredStake),
```

Comment:
I'm putting the percentage in the db so the staking banner knows whether to show itself or not (since it lives outside of the StakingWidget provider). (It doesn't really cut down on calculations on the client because we still need to calculate percentages for optimistic ui purposes.)